src/key_info_managers/mod.rs

```diff
@@ -144,4 +144,41 @@ pub trait ManageKeyInfo {
     ///
     /// Returns an error as a String if there was a problem accessing the Key Info Manager.
     fn exists(&self, key_triple: &KeyTriple) -> Result<bool, String>;
+
+    /// Returns a Vec of the KeyInfo objects corresponding to the given application name and
+    /// provider ID.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error as a String if there was a problem accessing the Key Info Manager.
+    fn list_key_info(
```

Comment:
I think I would prefer having this method outside of the trait so that it can not be overwritten:
```rust
fn list_key_info(
    &impl ManageKeyInfo,
    app_name: &ApplicationName,
    provider_id: ProviderID,
)
```